graph/patterns/navigation-property.md

```diff
@@ -0,0 +1,185 @@
+# Navigation Property
+
+Microsoft Graph API Design Pattern
+
+*A navigation property is used to identify a relationship between resources.*
+
+## Problem
+--------
+
+It is often valuable to represent a relationship between resources in an API. This may be a many-to-one or a one-to-many relationship. 
+
+Relationships between resources are often implicitly represented by a property contained in one of the resources that identifies the other related resource. Usually that information returned in a representation as an id value and the property is named using a convention that identifies the target type of related resource. e.g. userId 
+
+In many-to-one relationships, for a client to access the related resource it must request the primary resource, read the id value of the related resource and then construct a URL to the related resource using the Id value and knowledge of how the property name maps to the related resource. This requires at least two round trips and requires the client know how to construct the URL to the related resource.
+
+For both many-to-one and one-to-many relationships in order to retrieve information from both resources on both sides of the relationship requires at minimum two requests.
+
+Requiring two round trips to access this information is inefficient for some applications and the lack of a formally described relationship limits the ability for tooling to take advantage of the relationship to improve developer experience.
+
+## Solution
+--------
+
+Navigation properties are an [OData convention](https://docs.microsoft.com/en-us/odata/webapi/model-builder-untyped#navigation-property) that allows an API designer to describe a special kind of property in a model that references an related entity. In the HTTP API this property name translates to a path segment that can be appended to the URL of the primary resource in order to access a representation of the related resource. This prevents the client from needing to know any additional information on how to construct the URL to the related resource and the client does not need to retrieve the primary resource if it is only interested in the related resource.  It is the responsibility of the API implementation to determine the Id of the related resource and return the representation of the related entity.
+
+e.g. /user/{userId}/manager  # many-to-one relationship
+     /user/{userId}/messages # one-to-many relationship
+
+Additionally, using the OData Expand query parameter, related entities can be transcluded into the primary entity so both can be retrieved in a single round trip.
+
+## Issues and Considerations
+-------------------------
+
+In the current Microsoft Graph implementation, support for navigation properties is limited to entities within the same backend service or the user entity.  
+ 
```

Comment:
It seems that `user` entity violates this rule. may be it's just not recommended ?